Mathlib/RingTheory/Polynomial/Eisenstein/Distinguished.lean

```diff
@@ -43,22 +43,43 @@ lemma map_eq_X_pow {f : R[X]} {I : Ideal R} (distinguish : f.IsDistinguishedAt I
 @[deprecated (since := "2025-04-27")]
 alias _root_.IsDistinguishedAt.map_eq_X_pow := map_eq_X_pow
 
-lemma degree_eq_order_map {I : Ideal R} (f : PowerSeries R)
-    (h : R⟦X⟧) {g : R[X]} (distinguish : g.IsDistinguishedAt I)
-    (nmem : PowerSeries.constantCoeff R h ∉ I)
-    (eq : f = g * h) : g.degree = (f.map (Ideal.Quotient.mk I)).order := by
+section degree_eq_order_map
+
+variable {I : Ideal R} (f h : R⟦X⟧) {g : R[X]} (distinguish : g.IsDistinguishedAt I)
+  (nmem : PowerSeries.constantCoeff R h ∉ I) (eq : f = g * h)
+include distinguish nmem eq
+
+lemma map_ne_zero_of_eq_mul : f.map (Ideal.Quotient.mk I) ≠ 0 := fun H ↦ by
+  have mapf : f.map (Ideal.Quotient.mk I) = (Polynomial.X ^ g.natDegree : (R ⧸ I)[X]) *
+      h.map (Ideal.Quotient.mk I) := by
+    simp only [← map_eq_X_pow distinguish, Polynomial.polynomial_map_coe, eq, _root_.map_mul]
+  apply_fun PowerSeries.coeff _ g.natDegree at H
+  simp [mapf, PowerSeries.coeff_X_pow_mul', eq_zero_iff_mem, nmem] at H
+
+lemma degree_eq_coe_lift_order_map : g.degree = (f.map (Ideal.Quotient.mk I)).order.lift
```

Comment:
`g` cannot be zero, right? Because it is distinguished (and as you prove `R` is not trivial). In this case you can make the statement even more precise using `WithBot.untop` on the `degree`.

Reply:
Then it cannot be used in `rw` since naturally we will see `g.degree` everywhere but not `g.degree.unbot ...`.